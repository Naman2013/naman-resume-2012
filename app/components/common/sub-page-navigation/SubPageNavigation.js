/***********************************
 * V4 Sub Page Navigation
 *  Component that will navigate through the different containers of a page
 *  Currently used by V4 Object Details and V4 Guide Details
 ***********************************/

import React, { Fragment, useEffect } from 'react';
import { Link } from 'react-router';
import PropTypes from 'prop-types';
import style from './SubPageNavigation.style';

const SubPageNavigation = ({ items, locationPath }) => {
  useEffect(() => {
    const el = document.querySelector('#navigationId .subnav-active');
    el.scrollIntoView({ inline: 'start' });
  }, [locationPath]);

  return (
    <Fragment>
      <ul className="subnav" id="navigationId">
        {items.map(item => (
          <li className="item" key={item.link}>
            <Link
              activeClassName="subnav-active"
              className="subnav-link"
              to={item.link}
            >
              {item.title}
            </Link>
          </li>
        ))}
      </ul>
      <style jsx global>
        {style}
      </style>
    </Fragment>
  );
};

SubPageNavigation.defaultProps = {
  items: [],
};

SubPageNavigation.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired,
    })
  ),
};

export default SubPageNavigation;
