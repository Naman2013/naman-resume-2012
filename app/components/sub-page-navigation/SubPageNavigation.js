/***********************************
* V4 Sub Page Navigation
*  Component that will navigate through the different containers of a page
*  Currently used by V4 Object Details and V4 Guide Details
***********************************/

import React, { Component } from 'react';
import { Link } from 'react-router';
import PropTypes from 'prop-types';

const SubPageNavigation = ({ items }) => (
  <div className="navigation-root">
    <ul className="navigation">
      {
        items.map(item => (
          <li className="item" key={item.link}>
            <Link
              activeClassName="active"
              to={item.link}
            >
              <span className="link">
                {item.title}
              </span>
            </Link>
          </li>
        ))
      }
    </ul>
    <style jsx>{`

      .navigation {
        display: flex;
        flex-direction: row;
      }

      .item {
        list-style-type: none;
        padding: 0 25px;
      }

      :global(.active),
      :global(.active):link,
      :global(.active):active,
      :global(.active):visited {
        font-weight: normal;
        text-transform: none;
        display: block;
        text-decoration: none;
      }

      :global(.active) .link {
        font-weight: bold;
      }
    `}</style>
  </div>
);

SubPageNavigation.defaultProps = {
  items: [],
};

SubPageNavigation.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  })),
};

export default SubPageNavigation;
