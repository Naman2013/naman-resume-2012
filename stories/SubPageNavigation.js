import React from 'react';
import { storiesOf } from '@storybook/react';
import SubPageNavigation from '../app/components/common/sub-page-navigation';

const generateNavItems = (objectId) => ([
  {
    title: 'Neque',
    link:`/object-details/${objectId}/Neque`
  },
  {
    title: 'porro',
    link:`/object-details/${objectId}/porro`
  },
  {
    title: 'quisquam',
    link:`/object-details/${objectId}/quisquam`
  },
  {
    title: 'est',
    link:`/object-details/${objectId}/est`
  },
  {
    title: 'qui',
    link:`/object-details/${objectId}/qui`
  },
  {
    title: 'dolorem',
    link:`/object-details/${objectId}/dolorem`
  },
  {
    title: 'ipsum',
    link:`/object-details/${objectId}/ipsum`
  },
]);

storiesOf('SubPageNavigation', module)
  .add('Default', () => (
    <SubPageNavigation items={generateNavItems(1)} />
  ));
