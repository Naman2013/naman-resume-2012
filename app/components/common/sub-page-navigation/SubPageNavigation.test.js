import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SubPageNavigation from './';

const NAV_ITEMS = [
  {
    title: 'Overview',
    link:`/object-details/overview`
  },
  {
    title: 'Ask an Astronomer',
    link:`/object-details/ask`
  }
];

it('should render nothing when no links', () => {
  const tree = renderer.create(
    <SubPageNavigation />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render a list of links', () => {
  const tree = renderer.create(
    <SubPageNavigation items={NAV_ITEMS} />,
  );
  const component = mount(<SubPageNavigation items={NAV_ITEMS} />);
  expect(tree.toJSON()).toMatchSnapshot();
  expect(component.find('a').length).toBe(2);
  expect(component.find('a').first().text()).toBe(NAV_ITEMS[0].title);
});
