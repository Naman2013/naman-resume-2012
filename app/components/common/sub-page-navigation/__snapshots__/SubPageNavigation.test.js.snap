// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of links 1`] = `
<div
  className="jsx-557664041 navigation-root"
>
  <ul
    className="jsx-557664041 navigation"
  >
    <li
      className="jsx-557664041 item"
    >
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="jsx-557664041 link"
        >
          Overview
        </span>
      </a>
    </li>
    <li
      className="jsx-557664041 item"
    >
      <a
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="jsx-557664041 link"
        >
          Ask an Astronomer
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render nothing when no links 1`] = `
<div
  className="jsx-557664041 navigation-root"
>
  <ul
    className="jsx-557664041 navigation"
  />
</div>
`;
